enhancements/sig-policy/122-expanded-hub-templates-access/README.md

```diff
@@ -0,0 +1,310 @@
+launching a goroutine that replaces the token before expiration. Note that if the token is provided in a token file,
+client-go will periodically check for a new token in the file and use that for API requests. Taking advantage of this
+prevents the Governance Policy Propagator from needing to recreate the `TemplateResolver` instance when the token
+changes. See an example function to get a service account token that gets automatically refreshed:
+
+```go
+func (r *ReplicatedPolicyReconciler) getToken(ctx context.Context, namespace string, saName string) (string, error) {
+	var expiration int64 = 7200
+
+	req := &authv1.TokenRequest{
+		Spec: authv1.TokenRequestSpec{
+			// TODO: Is this always the correct value?
+			Audiences: []string{"https://kubernetes.default.svc.cluster.local"},
+			// 2 hours
+			ExpirationSeconds: &expiration,
+		},
+	}
+
+	returnedTokenReq, err := r.HubClient.CoreV1().ServiceAccounts(namespace).CreateToken(
+		ctx, saName, req, metav1.CreateOptions{},
+	)
+	if err != nil {
+		return "", err
+	}
+
+	tokenFile, err := os.CreateTemp("", fmt.Sprintf("token-%s.%s-*", namespace, saName))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = tokenFile.Write([]byte(returnedTokenReq.Status.Token))
+	if err != nil {
+		return "", err
+	}
+
+	tokenFilePath, err := filepath.Abs(tokenFile.Name())
+	if err != nil {
+		return "", err
+	}
+
+	expirationTimestamp := returnedTokenReq.Status.ExpirationTimestamp
+
+	go func() {
+		defer func() {
+			err := os.Remove(tokenFilePath)
+			if err != nil {
+				log.Error(err, "Failed to clean up the service account token file", "serviceAccount", saName, "namespace", namespace, "path", tokenFilePath)
+			}
+		}()
+
+		for {
+			deadlineCtx, deadlineCtxCancel := context.WithDeadline(ctx, expirationTimestamp.Add(-10*time.Minute))
+
+			<-deadlineCtx.Done()
+			if !errors.Is(deadlineCtx.Err(), context.DeadlineExceeded) {
+				// This really does nothing but this satisfies the linter that cancel is called.
+				deadlineCtxCancel()
+
+				return
+			}
+
+			returnedTokenReq, err := r.HubClient.CoreV1().ServiceAccounts(namespace).CreateToken(
+				ctx, saName, req, metav1.CreateOptions{},
+			)
+			if err != nil {
+				log.Error(err, "Failed to renew the token. Will retry in 5 seconds.", "serviceAccount", saName, "namespace", namespace)
+
+				time.Sleep(5 * time.Second)
+
+				continue
+			}
+
+			_, err = tokenFile.Write([]byte(returnedTokenReq.Status.Token))
+			if err != nil {
+				log.Error(err, "Failed to write the token. Will retry in 5 seconds.", "serviceAccount", saName, "namespace", namespace, "path", tokenFilePath)
+
+				time.Sleep(5 * time.Second)
+
+				continue
+			}
+
+			expirationTimestamp = returnedTokenReq.Status.ExpirationTimestamp
+
+			// This really does nothing but this satisfies the linter that cancel is called.
+			deadlineCtxCancel()
+		}
+	}()
+
+	return tokenFilePath, nil
+}
+```
+
+### Risks and Mitigation
```

Comment:
There could be some cases where customers have policies deployed to namespaces that have serviceaccounts already and they could take advantage of these permissions to gain some extra privileges they didn't have before.  The best practice of not mixing policies with workloads would be the mitigation.